Group_By.py

```diff
@@ -0,0 +1,27 @@
+Persons = [
+    {"Name": "Ali", "FatherName": "Khan", "Age": 30, "Location": "Gulberg"},
+    {"Name": "Sara", "FatherName": "Ahmed", "Age": 27, "Location": "Defence"},
+    {"Name": "Usman", "FatherName": "Arshad", "Age": 32, "Location": "Shahdra"},
+    {"Name": "Amina", "FatherName": "Hussain", "Age": 22, "Location": "Model Town"},
+    {"Name": "Ahmed", "FatherName": "Rasheed", "Age": 28, "Location": "Gulberg"},
+    {"Name": "Zara", "FatherName": "Siddique", "Age": 31, "Location": "Defence"},
+    {"Name": "Hassan", "FatherName": "Malik", "Age": 26, "Location": "Gulberg"},
+    {"Name": "Fatima", "FatherName": "Rizvi", "Age": 29, "Location": "Shahdra"},
+    {"Name": "Bilal", "FatherName": "Azam", "Age": 24, "Location": "Model Town"},
+    {"Name": "Hina", "FatherName": "Ali", "Age": 33, "Location": "Defence"}
+]
+
+
+def GroupByFunc(list):
```

Comment:
we need to pass 2 parameters list and key ( key on which we want to group so the function is generic ).
We need to do same function using reduce function